test/acquisition/multi_objective/test_monte_carlo.py

```diff
@@ -556,6 +570,11 @@ def setUp(self):
             device=self.device,
         )
 
+    @mock.patch.object(
+        IdentityMCMultiOutputObjective,
+        "_verify_output_shape",
+        mock.PropertyMock(return_value=False),
+    )
```

Comment:
I went with this blanket patch to avoid having to repeat it in many places. NEHVI in general requires 2 different posterior shapes (only for baseline and baseline + current X), which does not work when using `MockPosterior`. However, it seems that this blanket patch ended up hiding some bugs that were only caught thanks to the tutorials. Implementing a new mock posterior that adapts to the input shape is probably a better way to go here.